diff --git a/axi.core b/axi.core
index 3b4cc5bd..169f4cf1 100644
--- a/axi.core
+++ b/axi.core
@@ -71,48 +71,14 @@ filesets:
       - src/axi_to_mem_banked.sv
       - src/axi_to_mem_interleaved.sv
       - src/axi_to_mem_split.sv
-      - src/axi_chan_compare.sv
-      - src/axi_dumper.sv
-      - src/axi_sim_mem.sv
-      - src/axi_test.sv
       # Level 5
       - src/axi_xbar.sv
       # Level 6
       - src/axi_xp.sv
     file_type : systemVerilogSource
     depend :
-      - ">=pulp-platform.org::common_cells:1.31.1"
-  benchs:
-    files:
-      - test/tb_axi_dw_pkg.sv
-      - test/tb_axi_xbar_pkg.sv
-      - test/axi_synth_bench.sv
-      - test/tb_axi_addr_test.sv
-      - test/tb_axi_atop_filter.sv
-      - test/tb_axi_bus_compare.sv
-      - test/tb_axi_cdc.sv
-      - test/tb_axi_delayer.sv
-      - test/tb_axi_dw_downsizer.sv
-      - test/tb_axi_dw_upsizer.sv
-      - test/tb_axi_fifo.sv
-      - test/tb_axi_isolate.sv
-      - test/tb_axi_iw_converter.sv
-      - test/tb_axi_lite_dw_converter.sv
-      - test/tb_axi_lite_mailbox.sv
-      - test/tb_axi_lite_regs.sv
-      - test/tb_axi_lite_to_apb.sv
-      - test/tb_axi_lite_to_axi.sv
-      - test/tb_axi_lite_xbar.sv
-      - test/tb_axi_modify_address.sv
-      - test/tb_axi_serializer.sv
-      - test/tb_axi_sim_mem.sv
-      - test/tb_axi_slave_compare.sv
-      - test/tb_axi_to_axi_lite.sv
-      - test/tb_axi_to_mem_banked.sv
-      - test/tb_axi_xbar.sv
-    file_type : systemVerilogSource
-    depend :
-      - ">=pulp-platform.org::common_verification:0.2.5"
+      - ">=pulp-platform.org::common_cells:1.29.0"
+
 
 generators:
   axi_intercon_gen:
@@ -174,7 +140,7 @@ targets:
   default:
     filesets : [rtl]
   sim: &sim
-    filesets : [rtl,benchs]
+    filesets : [rtl]
     description: Simulate the design
     toplevel: tb_axi_delayer
   sim_dw_downsizer:
