diff --git a/src/axi_demux_simple.sv b/src/axi_demux_simple.sv
index c7ba39f..1306e3d 100644
--- a/src/axi_demux_simple.sv
+++ b/src/axi_demux_simple.sv
@@ -422,14 +422,14 @@ module axi_demux_simple #(
         // AW channel
         mst_reqs_o[i].aw       = slv_req_i.aw;
         mst_reqs_o[i].aw_valid = 1'b0;
-        if (aw_valid && (slv_aw_select_i == i)) begin
+        if (aw_valid && (slv_aw_select_i == select_t'(i))) begin
           mst_reqs_o[i].aw_valid = 1'b1;
         end
 
         //  W channel
         mst_reqs_o[i].w       = slv_req_i.w;
         mst_reqs_o[i].w_valid = 1'b0;
-        if (w_select_valid && (w_select == i)) begin
+        if (w_select_valid && (w_select == select_t'(i))) begin
           mst_reqs_o[i].w_valid = slv_req_i.w_valid;
           slv_resp_o.w_ready           = mst_resps_i[i].w_ready;
           w_cnt_down            = slv_req_i.w_valid & mst_resps_i[i].w_ready & slv_req_i.w.last;
@@ -441,7 +441,7 @@ module axi_demux_simple #(
         // AR channel
         mst_reqs_o[i].ar       = slv_req_i.ar;
         mst_reqs_o[i].ar_valid = 1'b0;
-        if (ar_valid && (slv_ar_select_i == i)) begin
+        if (ar_valid && (slv_ar_select_i == select_t'(i))) begin
           mst_reqs_o[i].ar_valid = 1'b1;
         end
 
diff --git a/src/axi_isolate.sv b/src/axi_isolate.sv
index b5596e7..76a32ad 100644
--- a/src/axi_isolate.sv
+++ b/src/axi_isolate.sv
@@ -331,7 +331,7 @@ module axi_isolate_inner #(
     unique case (state_ar_q)
       Normal: begin
         // cut handshake if counter capacity is reached
-        if (pending_ar_q >= NumPending) begin
+        if (pending_ar_q >= cnt_t'(NumPending)) begin
           mst_req_o.ar_valid  = 1'b0;
           slv_resp_o.ar_ready = 1'b0;
           if (isolate_i) begin
diff --git a/src/axi_lite_to_axi.sv b/src/axi_lite_to_axi.sv
index d34190f..0f1a966 100644
--- a/src/axi_lite_to_axi.sv
+++ b/src/axi_lite_to_axi.sv
@@ -33,7 +33,7 @@ module axi_lite_to_axi #(
   output axi_req_t        mst_req_o,
   input  axi_resp_t       mst_resp_i
 );
-  localparam int unsigned AxiSize = axi_pkg::size_t'($unsigned($clog2(AxiDataWidth/8)));
+  localparam axi_pkg::size_t AxiSize = axi_pkg::size_t'($unsigned($clog2(AxiDataWidth/8)));
 
   // request assign
   assign mst_req_o = '{
diff --git a/src/axi_to_detailed_mem.sv b/src/axi_to_detailed_mem.sv
index d0933fe..49aee32 100644
--- a/src/axi_to_detailed_mem.sv
+++ b/src/axi_to_detailed_mem.sv
@@ -851,7 +851,7 @@ module mem_stream_to_banks_detailed #(
 
     // Can issue another request if the counter is not at its limit or a response is delivered in
     // the current cycle.
-    assign cnt_req_ready = (cnt_q < MaxTrans) | (rvalid_o & rready_i);
+    assign cnt_req_ready = (cnt_q < cnt_t'(MaxTrans)) | (rvalid_o & rready_i);
 
     // Control request and memory request interface handshakes.
     assign gnt_o = mem_req_ready & cnt_req_ready;

diff --git a/src/axi_atop_filter.sv b/src/axi_atop_filter.sv
index 55ad491..3619fd8 100644
--- a/src/axi_atop_filter.sv
+++ b/src/axi_atop_filter.sv
@@ -120,7 +120,7 @@ module axi_atop_filter #(
 
       W_FEEDTHROUGH: begin
         // Feed AW channel through if the maximum number of outstanding bursts is not reached.
-        if (complete_w_without_aw_downstream || (w_cnt_q.cnt < AxiMaxWriteTxns)) begin
+        if (complete_w_without_aw_downstream || (32'(w_cnt_q.cnt) < AxiMaxWriteTxns)) begin
           mst_req_o.aw_valid  = slv_req_i.aw_valid;
           slv_resp_o.aw_ready = mst_resp_i.aw_ready;
         end
